lib/search.rb

```diff
@@ -260,6 +261,10 @@ def valid?
     @valid
   end
 
+  def use_full_page_limit
+    @opts[:search_type] == :full_page || Topic === @search_context
```

Comment:
I see in the tests that we passed in `search_type == :header`. Does that matter here?